syntax = "proto3";

package post;

import "google/protobuf/empty.proto";

service PostController {
    rpc List(PostListRequest) returns (stream Post) {}
    rpc Create(Post) returns (Post) {}
    rpc Retrieve(PostRetrieveRequest) returns (Post) {}
    rpc Update(Post) returns (Post) {}
    rpc Destroy(Post) returns (google.protobuf.Empty) {}
}

message Post {
    int32 id = 1;
    string title = 2;
    string content = 3;
    string image = 4;
    bool is_active = 5;
    string created = 6;
    string updated = 7;
    int32 category = 8;
    int32 author = 9;
}

message PostListRequest {
}

message PostRetrieveRequest {
    int32 id = 1;
}
